### PR TITLE
Use RTE for product description.

### LINKED ISSUE #11039: Integrate ActionText by default
Integrate [action_text ](https://edgeguides.rubyonrails.org/action_text_overview.html) by default in products form in admin

### PR DESCRIPTION
**Pull request by MatthewKennedy**
- Tests to write

Note: Moved raw frontend product description code outside of shot/ long description behaviour, truncating HTML results in broken HTML.

Fixes https://github.com/spree/spree/issues/11039

### CODE REVIEW

File: backend/app/views/spree/admin/products/_form.html.erb [hunk 1]
```diff
@@ -21,7 +21,7 @@
       <div data-hook="admin_product_form_description">
         <%= f.field_container :description, class: ['form-group'] do %>
           <%= f.label :description, Spree.t(:description) %>
-          <%= f.text_area :description, { rows: "#{unless @product.has_variants? then '20' else '13' end}", class: 'form-control' } %>
+          <%= f.text_area :description, { rows: "#{unless @product.has_variants? then '29' else '30' end}", class: "form-control #{"spree-rte" if Spree::Config[:use_rte_for_product_description] }" } %>
```
Comment (damianlegawiec) [inline backend/app/views/spree/admin/products/_form.html.erb:24]:
```suggestion
          <%= f.text_area :description, { rows: "#{unless @product.has_variants? then '29' else '30' end}", class: "form-control #{"spree-rte" if product_wysywig_editor_enabled? }" } %>
```

```ruby
def product_wysywig_editor_enabled?
  Spree::Backend::Config[:product_wysywig_editor_enabled]
end
```

File: core/app/helpers/spree/products_helper.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ def used_variants_options(variants, product)
 
     # converts line breaks in product description into <p> tags (for html display purposes)
     def product_description(product)
-      description = if Spree::Config[:show_raw_product_description]
+      description = if Spree::Config[:show_raw_product_description] || Spree::Config[:use_rte_for_product_description]
```
[the remark this answers is not in the document]
Comment (MatthewKennedy) [reply core/app/helpers/spree/products_helper.rb:49]: Would it not be better to let the RTE take care of all the HTML formatting when in use?

[the remark this answers is not in the document]
Comment (damianlegawiec) [reply core/app/helpers/spree/products_helper.rb:49]: @MatthewKennedy you're right, I was under the impression that `show_raw_product_description` is enabled by default, but it isn't

[the remark this answers is not in the document]
Comment (MatthewKennedy) [reply core/app/helpers/spree/products_helper.rb:49]: @damianlegawiec,  That was one question I wanted to ask, do you want RTE on by default?

[the remark this answers is not in the document]
Comment (damianlegawiec) [reply core/app/helpers/spree/products_helper.rb:49]: @MatthewKennedy yes, it should be set by default, however, we need to add a note about in releases notes and instructions how to disable it in release notes and upgrade guide

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**